### ISSUE TITLE
Sensu asset will not update using new `build` block syntax

### ISSUE THREAD

**Issue by mdelaney** [jtopjian/terraform-provider-sensu#34]
In the [doc page](https://github.com/jtopjian/terraform-provider-sensu/blob/cef5b67b9cc38def903e7513fac12c940e8b1a9f/docs/resources/sensu_asset.md) for assets the basic example shows the new way to define an asset using the build block. It seems that assets do not get updated when using the `build` block.

terraform plan and apply both correctly show what should be updated but when you run apply the resource is unchanged.

Here is our example resource
```terraform
resource "sensu_asset" "sensu-plugins-foo" {
  name = "sensu-plugins-foo"

  build {
    sha512 = "fa6b95de15bd598024305b2c105f625a9f3275cc1e3b8c195b2731d903f7c1308cdcc5c2f14b8a9662566595d32180973ae9bcf9edfc8665799f69f1481858df"
    url    = "http://here.foo/sensu-plugin-foo_0.19.0_linux_amd64.tar.gz"
    filters = [
      "entity.system.os == 'linux'",
      "entity.system.arch == 'amd64'",
    ]
  }
}
```
If we update the version and hash we see the expected changes when running terraform
```
  # sensu_asset.sensu-plugins-foo will be updated in-place
  ~ resource "sensu_asset" "sensu-plugins-foo" {
        annotations = {}
        filters     = []
        headers     = {}
        id          = "sensu-plugins-foo"
        labels      = {}
        name        = "sensu-plugins-foo"
        namespace   = "infra"

      ~ build {
            filters = [
                "entity.system.os == 'linux'",
                "entity.system.arch == 'amd64'",
            ]
            headers = {}
          ~ sha512  = "2f5ade30cf2db44556d84e21860142bf211dda1d7b50105cfba448731df0a861d4eaeafd292813d15cabca85b23a0c4093b3c6e608cef5ac9750a5730fcc361c" -> "f
a6b95de15bd598024305b2c105f625a9f3275cc1e3b8c195b2731d903f7c1308cdcc5c2f14b8a9662566595d32180973ae9bcf9edfc8665799f69f1481858df"
          ~ url     = "http://here.foo/sensu-plugin-foo_0.19.0_linux_amd64.tar.gz" -> "http://here.foo/sensu-plugin-foo_0.20.0_linux_amd64.tar.gz"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
And when the change is applied, terraform reports success (1 item changed).

When we look at the assets defined with `sensuctl` we still have the old asset.

If we remove the `build` block and go back to the deprecated syntax everything works as expected. 

---

When time allows we'll dig into this further and look at opening a PR to fix but figured I'd open the issue in-case someone else has time to dig in first :-)

**Comment by jtopjian** [comment]
I suppose it would help if the Update function actually updated builds 🙂 

https://github.com/jtopjian/terraform-provider-sensu/blob/master/sensu/resource_asset.go#L150

I'll get this fixed up in the next day or two. Sorry about that!

**Comment by jtopjian** [comment]
@mdelaney This should now be fixed and available in v0.10.1 which I just released. Let me know if you have any questions or run into any other issues.

**Comment by mdelaney** [comment]
Much appreciated @jtopjian! We'll try it out Monday.

Thanks again for the fast turnaround on this!
